controller/uninstall_controller.go

```diff
@@ -382,6 +382,48 @@ func (c *UninstallController) deleteCRDs() (bool, error) {
 		return true, c.deleteReplicas(replicas)
 	}
 
+	// Unset backup target to prevent the remote backup target
+	// backup volume config, and backup config and it's data
+	// being deleted during uninstall process.
+	// TODO:
+	//   Remove the setting CRs and add OwnerReferences on BackupTarget CR.
+	//   After that when deleting setting CRs, the default BackupTarget CR
+	//   be cascading deleted automatically.
+	targetSetting, err := c.ds.GetSetting(types.SettingNameBackupTarget)
+	if err != nil {
+		return true, err
+	}
+	if targetSetting.Value != "" {
+		targetSetting.Value = ""
+		if _, err := c.ds.UpdateSetting(targetSetting); err != nil {
+			return true, err
+		}
+	}
+
+	// Waits the BackupVolume CRs be clean up by backup_target_controller
+	if backupVolumes, err := c.ds.ListBackupVolumes(); err != nil {
+		return true, err
+	} else if len(backupVolumes) > 0 {
+		c.logger.Infof("Found %d backupvolumes remaining", len(backupVolumes))
+		return true, nil
+	}
+
+	// Waits the Backup CRs be clean up by backup_volume_controller
+	if backups, err := c.ds.ListBackups(); err != nil {
+		return true, err
+	} else if len(backups) > 0 {
+		c.logger.Infof("Found %d backups remaining", len(backups))
+		return true, nil
+	}
+
+	// Delete BackupTarget CRs after the BackupVolume/Backup CR be clean up
+	if backupTargets, err := c.ds.ListBackupTargets(); err != nil {
+		return true, err
+	} else if len(backupTargets) > 0 {
+		c.logger.Infof("Found %d backuptargets remaining", len(backupTargets))
+		return true, c.deleteBackupTargets(backupTargets)
+	}
+
```

Comment:
We don't need to delete and wait for BackupTarget CR. It will be gone with the CRD removal.